diff -Nur a/src/app/mpd-client/Makefile.alone b/src/app/mpd-client/Makefile.alone
--- a/src/app/mpd-client/Makefile.alone	2020-01-19 21:03:51.000000000 +0800
+++ b/src/app/mpd-client/Makefile.alone	2020-02-18 21:32:48.580001518 +0800
@@ -1,6 +1,9 @@
 DEBUG_OPTIONS = -g -Wall
 OPTIMIZE = -O3
 
+UCLIBCXX_INCLUDES = -I/usr/include/uClibc\+\+
+UCLIBCXX_LIBS = -luClibc\+\+
+
 SINGLE_C_FILE_PROGRAMS =
 SINGLE_CXX_FILE_PROGRAMS =
 SINGLE_FILE_PROGRAMS = $(SINGLE_C_FILE_PROGRAMS) $(SINGLE_CXX_FILE_PROGRAMS)
@@ -46,8 +49,9 @@
 endif
 
 CFLAGS += -I../../include -I../../libs/include
-CXXFLAGS += -I../../include -I../../libs/include
-LDFLAGS += -L../../libs/lbk -llbk
+CXXFLAGS += -I../../include -I../../libs/include $(UCLIBCXX_INCLUDES)
+LDFLAGS += -L../../libs/lbk -llbk $(UCLIBCXX_LIBS)
+
 
 all: targets $(TARGET)
 	ln -sf ./LBMPClient _APP_
diff -Nur a/src/app/panel/Makefile.alone b/src/app/panel/Makefile.alone
--- a/src/app/panel/Makefile.alone	2019-07-21 02:46:44.000000000 +0800
+++ b/src/app/panel/Makefile.alone	2020-02-18 21:32:48.540001518 +0800
@@ -1,6 +1,9 @@
 DEBUG_OPTIONS = -g -Wall
 OPTIMIZE = -O3
 
+UCLIBCXX_INCLUDES = -I/usr/include/uClibc\+\+
+UCLIBCXX_LIBS = -luClibc\+\+
+
 SINGLE_C_FILE_PROGRAMS =
 SINGLE_CXX_FILE_PROGRAMS =
 SINGLE_FILE_PROGRAMS = $(SINGLE_C_FILE_PROGRAMS) $(SINGLE_CXX_FILE_PROGRAMS)
@@ -30,8 +33,8 @@
 LD = ${CROSS_COMPILE}ld
 
 CFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE)
-CXXFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE)
-LDFLAGS =
+CXXFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(UCLIBCXX_INCLUDES)
+LDFLAGS = $(UCLIBCXX_LIBS)
 
 ifeq ($(IS_BEOS_PLATFORM),0)
 	CFLAGS += $(shell lite-beapi-config --cflags)
diff -Nur a/src/cmd/lbk/Makefile.alone b/src/cmd/lbk/Makefile.alone
--- a/src/cmd/lbk/Makefile.alone	2019-07-21 02:46:44.000000000 +0800
+++ b/src/cmd/lbk/Makefile.alone	2020-02-18 21:32:48.360001517 +0800
@@ -1,6 +1,9 @@
 DEBUG_OPTIONS = -g -Wall
 OPTIMIZE = -O3
 
+UCLIBCXX_INCLUDES = -I/usr/include/uClibc\+\+
+UCLIBCXX_LIBS = -luClibc\+\+
+
 C_PROGRAM := 
 CXX_PROGRAM := 
 
@@ -52,8 +55,8 @@
 endif
 
 CFLAGS += -I../../include -I../../libs/include
-CXXFLAGS += -I../../include -I../../libs/include
-LDFLAGS += -L../../libs/lbk -llbk
+CXXFLAGS += -I../../include -I../../libs/include $(UCLIBCXX_INCLUDES)
+LDFLAGS += -L../../libs/lbk -llbk $(UCLIBCXX_LIBS)
 
 ifeq ($(C_PROGRAM), )
 CFLAGS += -DCMD_ALL_IN_ONE
diff -Nur a/src/libs/lbk/Makefile.alone b/src/libs/lbk/Makefile.alone
--- a/src/libs/lbk/Makefile.alone	2020-01-19 21:17:00.000000000 +0800
+++ b/src/libs/lbk/Makefile.alone	2020-02-18 22:57:04.932806510 +0800
@@ -1,6 +1,9 @@
 DEBUG_OPTIONS = -g -Wall
 OPTIMIZE = -O3
 
+UCLIBCXX_INCLUDES = -I/usr/include/uClibc\+\+
+UCLIBCXX_LIBS = -luClibc\+\+
+
 SINGLE_C_FILE_PROGRAMS =
 SINGLE_CXX_FILE_PROGRAMS =
 SINGLE_FILE_PROGRAMS = $(SINGLE_C_FILE_PROGRAMS) $(SINGLE_CXX_FILE_PROGRAMS)
@@ -19,8 +22,13 @@
 endif
 
 ADD_ONS_DIRS = \
-	generic \
+	generic
+
+OPENWRT_FOUND = $(shell cat /etc/os-release | grep 'OpenWrt')
+ifeq ($(OPENWRT_FOUND), )
+ADD_ONS_DIRS += \
 	vpd
+endif
 
 ifeq ($(shell uname),Linux)
 ADD_ONS_DIRS += \
@@ -60,8 +68,8 @@
 LD = ${CROSS_COMPILE}ld
 
 CFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE)
-CXXFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE)
-LDFLAGS =
+CXXFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(UCLIBCXX_INCLUDES)
+LDFLAGS = $(UCLIBCXX_LIBS)
 
 SO_CFLAGS = -fPIC
 SO_LDFLAGS =
diff -Nur a/src/libs/lbk/add-ons/generic/Makefile.alone b/src/libs/lbk/add-ons/generic/Makefile.alone
--- a/src/libs/lbk/add-ons/generic/Makefile.alone	2020-01-19 21:17:00.000000000 +0800
+++ b/src/libs/lbk/add-ons/generic/Makefile.alone	2020-02-18 23:00:10.959205745 +0800
@@ -1,6 +1,9 @@
 DEBUG_OPTIONS = -g -Wall
 OPTIMIZE = -O3
 
+UCLIBCXX_INCLUDES = -I/usr/include/uClibc\+\+
+UCLIBCXX_LIBS = -luClibc\+\+
+
 SINGLE_C_FILE_PROGRAMS =
 SINGLE_CXX_FILE_PROGRAMS =
 SINGLE_FILE_PROGRAMS = $(SINGLE_C_FILE_PROGRAMS) $(SINGLE_CXX_FILE_PROGRAMS)
@@ -19,9 +22,9 @@
 CC = ${CROSS_COMPILE}gcc
 CXX = ${CROSS_COMPILE}g++
 LD = ${CROSS_COMPILE}ld
-CFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(shell lite-beapi-config --cflags) -fPIC -I../../../../include -I../../../include
-CXXFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(shell lite-beapi-config --cflags) -fPIC -I../../../../include -I../../../include
-LDFLAGS =  $(shell lite-beapi-config --libs) -L../.. -llbk
+CFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(shell lite-beapi-config --cflags) -fPIC -I../../../../include -I../../../include $(UCLIBCXX_INCLUDES)
+CXXFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(shell lite-beapi-config --cflags) -fPIC -I../../../../include -I../../../include $(UCLIBCXX_INCLUDES)
+LDFLAGS =  $(shell lite-beapi-config --libs) -L../.. -llbk $(UCLIBCXX_LIBS)
 
 all: targets $(TARGET)
 
diff -Nur a/src/libs/lbk/add-ons/lcd-st7735s/Makefile.alone b/src/libs/lbk/add-ons/lcd-st7735s/Makefile.alone
--- a/src/libs/lbk/add-ons/lcd-st7735s/Makefile.alone	2020-01-19 21:17:00.000000000 +0800
+++ b/src/libs/lbk/add-ons/lcd-st7735s/Makefile.alone	2020-02-18 21:32:48.190001517 +0800
@@ -1,6 +1,9 @@
 DEBUG_OPTIONS = -g -Wall
 OPTIMIZE = -O3
 
+UCLIBCXX_INCLUDES = -I/usr/include/uClibc\+\+
+UCLIBCXX_LIBS = -luClibc\+\+
+
 SINGLE_C_FILE_PROGRAMS =
 SINGLE_CXX_FILE_PROGRAMS =
 SINGLE_FILE_PROGRAMS = $(SINGLE_C_FILE_PROGRAMS) $(SINGLE_CXX_FILE_PROGRAMS)
@@ -20,8 +23,8 @@
 CXX = ${CROSS_COMPILE}g++
 LD = ${CROSS_COMPILE}ld
 CFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(shell lite-beapi-config --cflags) -fPIC -I../../../../include -I../../../include
-CXXFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(shell lite-beapi-config --cflags) -fPIC -I../../../../include -I../../../include
-LDFLAGS =  $(shell lite-beapi-config --libs) -L../.. -llbk
+CXXFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(shell lite-beapi-config --cflags) -fPIC -I../../../../include -I../../../include $(UCLIBCXX_INCLUDES)
+LDFLAGS =  $(shell lite-beapi-config --libs) -L../.. -llbk $(UCLIBCXX_LIBS)
 
 all: targets $(TARGET)
 
diff -Nur a/src/libs/lbk/add-ons/oled-ssd1306/Makefile.alone b/src/libs/lbk/add-ons/oled-ssd1306/Makefile.alone
--- a/src/libs/lbk/add-ons/oled-ssd1306/Makefile.alone	2020-01-19 21:17:00.000000000 +0800
+++ b/src/libs/lbk/add-ons/oled-ssd1306/Makefile.alone	2020-02-18 21:32:48.160001517 +0800
@@ -1,6 +1,9 @@
 DEBUG_OPTIONS = -g -Wall
 OPTIMIZE = -O3
 
+UCLIBCXX_INCLUDES = -I/usr/include/uClibc\+\+
+UCLIBCXX_LIBS = -luClibc\+\+
+
 SINGLE_C_FILE_PROGRAMS =
 SINGLE_CXX_FILE_PROGRAMS =
 SINGLE_FILE_PROGRAMS = $(SINGLE_C_FILE_PROGRAMS) $(SINGLE_CXX_FILE_PROGRAMS)
@@ -20,8 +23,8 @@
 CXX = ${CROSS_COMPILE}g++
 LD = ${CROSS_COMPILE}ld
 CFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(shell lite-beapi-config --cflags) -fPIC -I../../../../include -I../../../include
-CXXFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(shell lite-beapi-config --cflags) -fPIC -I../../../../include -I../../../include
-LDFLAGS =  $(shell lite-beapi-config --libs) -L../.. -llbk
+CXXFLAGS = $(DEBUG_OPTIONS) $(OPTIMIZE) $(shell lite-beapi-config --cflags) -fPIC -I../../../../include -I../../../include $(UCLIBCXX_INCLUDES)
+LDFLAGS =  $(shell lite-beapi-config --libs) -L../.. -llbk $(UCLIBCXX_LIBS)
 
 all: targets $(TARGET)
 
